pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix

```diff
@@ -0,0 +1,74 @@
+{ stdenv, fetchFromGitHub, yarn2nix, makeDesktopItem, electron, riot-web }:
+
+let
+  executableName = "riot-desktop";
+  version = "1.0.4";
+  riot-web-src = fetchFromGitHub {
+    owner = "vector-im";
+    repo = "riot-web";
+    rev = "v${version}";
+    sha256 = "152mi81miams5a7l9rd12bnf6wkd1r0lyicgr35r5fq0p6z7a4dk";
+  };
+
+in yarn2nix.mkYarnPackage rec {
+  name = "riot-desktop";
+  inherit version;
+
+  src = "${riot-web-src}/electron_app";
+
+  # The package manifest should be copied on each update of this package.
+  # > cp ${riot-web-src}/electron_app/package.json riot-desktop-package.json
+  packageJSON = ./riot-desktop-package.json;
+
+  # The dependency expression can be regenerated using nixos.yarn2nix with the following command:
+  # > yarn2nix --lockfile=${riot-web-src}/electron_app/yarn.lock > riot-desktop-yarndeps.nix
+  yarnNix = ./riot-desktop-yarndeps.nix;
+
+  installPhase = ''
+    # resources
+    mkdir -p "$out/share/riot"
+    ln -s '${riot-web}' "$out/share/riot/webapp"
+    cp -r '${riot-web-src}/origin_migrator' "$out/share/riot/origin_migrator"
+    cp -r '.' "$out/share/riot/electron"
+
+    # icons
+    for icon in $out/share/riot/electron/build/icons/*.png; do
+      mkdir -p "$out/share/icons/hicolor/$(basename $icon .png)/apps"
+      ln -s "$icon" "$out/share/icons/hicolor/$(basename $icon .png)/apps/riot.png"
+    done
+
+    # desktop item
+    mkdir -p "$out/share"
+    ln -s "${desktopItem}/share/applications" "$out/share/applications"
+
+    # executable wrapper
+    mkdir -p "$out/bin"
+    cat > "$out/bin/${executableName}" <<EOF
+    #!${stdenv.shell}
+    '${electron}/bin/electron' "$out/share/riot/electron" "$@"
+    EOF
+    chmod +x "$out/bin/${executableName}"
+  '';
+
+  desktopItem = makeDesktopItem {
```

Comment:
Do they have desktop file? There's one in the debian package so I'd it would be here.

Reply:
I installed Riot Desktop and it shows up in rofi's desktop-file-based launcher as `Riot (Matrix Client)`, so I assume the answer is yes, this package creates a desktop file

Reply:
The `.desktop` file present in the Debian package is generated by `electron-builder` which we aren't using here since it tries to connect to the internet to fetch stuff.

It's much simpler to create a wrapper script and the desktop item ourselves.

Reply:
> electron-builder

I see, can you link how this file would then be generated when using that?
It should be kept in sync if it ever were to change.

Reply:
Links and instructions added to the file.